providers/zohocrm/write.go

```diff
@@ -103,7 +104,7 @@ func constructWritePayload(payload any) (any, error) {
 	if !ok {
 		objectData, ok := payload.(map[string]any)
 		if !ok {
-			return nil, common.ErrBadRequest
+			return nil, fmt.Errorf("expecting either a json object or a list of objects, but received type '%T'", payload) //nolint:err113,lll
```

Comment:
This will give us linter error. 

You can either wrap `common.ErrBadRequest` or define new error such as

var errUnexpectedInputTyp = errors.New("unexpected input type") 

you can use one or the other. Don't use `fmt.Errorf` alone.